jwst/source_catalog/tests/test_source_catalog.py

```diff
@@ -122,3 +122,28 @@ def test_model_without_apcorr_data(nircam_model_without_apcorr):
     step = SourceCatalogStep(save_results=False)
     cat = step.run(nircam_model_without_apcorr)
     assert cat is None
+
+
+def test_input_model_reset(nircam_model):
+    """ Changes to input model data are made in SourceCatalogStep - make sure that
+        these changes are correctly reversed so the input model data/err arrays
+        remain unchanged after processing (to avoid copying datamodel),
+        and that the units are in MJy/sr before and after."""
+
+    original_data = nircam_model.data
+    original_data_unit = nircam_model.meta.bunit_data
+    original_err = nircam_model.err
+    original_err_unit = nircam_model.meta.bunit_err
+
+    assert(original_data_unit == original_err_unit == 'MJy/sr')
+
+    step = SourceCatalogStep(snr_threshold=0.5, npixels=5,
+                             bkg_boxsize=50, kernel_fwhm=2.0,
+                             save_results=False)
+
+    step.run(nircam_model)
+
+    assert(np.all(original_data == nircam_model.data))
+    assert(np.all(original_err == nircam_model.err))
```

Comment:
```suggestion
    assert_array_equal(original_data, nircam_model.data))
    assert_array_equal(original_err, nircam_model.err)
```

You will also need to add this line to the top imports:  `from numpy.testing import assert_array_equal`.

`assert_array_equal` is better because it compares NaNs.